src/v2i-hub/CARMAStreetsPlugin/src/JsonToJ3224SDSMConverter.cpp

```diff
@@ -0,0 +1,368 @@
+#include "JsonToJ3224SDSMConverter.h"
+
+namespace CARMAStreetsPlugin
+{
+
+    // Template to use when created shared pointer objects for optional data
+    template <typename T>
+    T *create_store_shared(std::vector<std::shared_ptr<void>> &shared_pointers)
+    {
+        auto obj_shared = std::make_shared<T>();
+        shared_pointers.push_back(obj_shared);
+        return obj_shared.get();
+    }
+
+    // Template for shared pointers with array elements
+    template <typename T>
+    T *create_store_shared_array(std::vector<std::shared_ptr<void>> &shared_pointers, int size)
+    {
+        std::shared_ptr<T[]> array_shared(new T[size]{0});
+        shared_pointers.push_back(array_shared);
+        return array_shared.get();
+    }
+
+
+    bool JsonToJ3224SDSMConverter::parseJsonString(const std::string &consumedMsg, Json::Value &sdsmJsonOut) const
+    {
+        const auto jsonLen = static_cast<int>(consumedMsg.length());
```

Comment:
Is this method at all unique to sdsms or is it just parsing strings into JSON values? If it is not unique maybe we can add a TODO comment to consolidate all the places parse strings into JSON values and try to reuse a single method. Maybe atleast for CARMA-Streets Plugin.